lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1017,25 +1060,30 @@
             const style = GetOption(options, "style", "string", ["decimal", "percent", "currency"], "decimal");
             nf.style = style;
             nf.formatterToUse = toEnum(NumberFormatStyle, _.toUpperCase(style));
+            const useCurrency = style === "currency";
 
-            let c = GetOption(options, "currency", "string", undefined, undefined);
-            if (c !== undefined && !IsWellFormedCurrencyCode(c)) {
-                platform.raiseInvalidCurrencyCode(c);
+            let currency = GetOption(options, "currency", "string", undefined, undefined);
+            if (currency !== undefined && !IsWellFormedCurrencyCode(currency)) {
+                platform.raiseInvalidCurrencyCode(currency);
+            } else if (currency === undefined && useCurrency) {
+                platform.raiseMissingCurrencyCode();
             }
 
-            let mnfdDefault, mxfdDefault;
-            if (style === "currency") {
-                if (c === undefined) {
-                    platform.raiseMissingCurrencyCode();
-                }
-
-                c = _.toUpperCase(c);
-                nf.currency = c;
+            let cDigits = 0;
+            if (useCurrency) {
+                currency = _.toUpperCase(currency);
+                nf.currency = currency;
+                cDigits = platform.currencyDigits(currency);
+            }
 
-                nf.currencyDisplay = GetOption(options, "currencyDisplay", "string", ["code", "symbol", "name"], "symbol");
-                nf.currencyDisplayToUse = toEnum(NumberFormatCurrencyDisplay, _.toUpperCase(nf.currencyDisplay));
+            let currencyDisplay = GetOption(options, "currencyDisplay", "string", ["code", "symbol", "name"], "symbol");
```

Comment:
>["code", "symbol", "name"] [](start = 82, length = 26)

worried about tainting on array literals like this? (Assuming that fix is coming later though...)

Reply:
Yeah, to get ready for windows we should probably take another once-over on the tainting tests